openedx/core/lib/block_structure/block_structure.py

```diff
@@ -66,11 +66,11 @@ def __init__(self, root_block_usage_key):
 
     def __iter__(self):
         """
-        The default iterator for a block structure is a topological
-        traversal since it's the more common case and we currently
-        need to support DAGs.
+        The default iterator for a block structure is get_block_keys()
+        since we need to filter blocks as a list.
+        A topological traversal can be used to support DAGs.
```

Comment:
@robrap Yes, a block_structure remains as a block_dag.  However, this change here is to change the default iterator (which is used only by the Course API serializer) to iterate through all the block keys in the structure.  All Block Transformers, on the other hand, do not use this iterator.  They will continue to be explicit about their traversal requirements: topological, post_order, etc.